xstate-chatbot/nodejs/src/machine/hospital-details.js

```diff
@@ -0,0 +1,1068 @@
+            dialog.sendMessage(context, message);
+          }),
+          on: {
+            USER_MESSAGE: 'process',
+          },
+        },
+        process: {
+          onEntry: assign((context, event) => {
+            if (event.message.type == 'text') {
+              const message = dialog.get_input(event, false);
+              context.slots.hospital.discharged_covid_patients_with_venti_l3 = message;
+              context.validMessage = true;
+            } else {
+              context.validMessage = false;
+            }
+          }),
+          always: [
+            {
+              cond: (context) => context.validMessage,
+              target: '#l3DeathswithVentilator',
+            },
+            {
+              target: 'error',
+            },
+          ],
+        },
+        error: {
+          onEntry: assign((context, event) => {
+            dialog.sendMessage(context, dialog.get_message(messages.l2TotalCOVIDdeaths.error, context.user.locale), false);
+          }),
+          always: 'prompt',
+        },
+      },
+    },//Total discharged COVID patients till date  L3 ICU (With Ventilator)
+    l3DeathswithVentilator: {
+      id: 'l3DeathswithVentilator',
+      initial: 'prompt',
+      states: {
+        prompt: {
+          onEntry: assign((context, event) => {
+            let message = dialog.get_message(messages.l3PreviousDeathswithVentilator.prompt, context.user.locale);
+            message += `\t::\t ${context.slots.previoushospitaldata.deaths_covid_patients_with_venti_l3}`;
+            message += `\n`;
+            message += `\n`;
+            message += dialog.get_message(messages.l3DeathswithVentilator.prompt, context.user.locale);
+            dialog.sendMessage(context, message);
+          }),
+          on: {
+            USER_MESSAGE: 'process',
+          },
+        },
+        process: {
+          onEntry: assign((context, event) => {
+            if (event.message.type == 'text') {
+              const message = dialog.get_input(event, false);
+              context.slots.hospital.deaths_covid_patients_with_venti_l3 = message;
+              context.validMessage = true;
+            } else {
+              context.validMessage = false;
+            }
+          }),
+          always: [
+            {
+              cond: (context) => context.validMessage,
+              target: '#l2UpdateHospitalDetails',
+            },
+            {
+              target: 'error',
+            },
+          ],
+        },
+        error: {
+          onEntry: assign((context, event) => {
+            dialog.sendMessage(context, dialog.get_message(messages.l2TotalCOVIDdeaths.error, context.user.locale), false);
+          }),
+          always: 'prompt',
+        },
+      },
+    },//Total COVID deaths till date  L3 ICU (With Ventilator)
+    l2l3Hospital: {
+      id: 'l2l3Hospital',
+      initial: 'process',
+      states: {
+        prompt: {
```

Comment:
No need to make 3 substates(prompt, process,error) for the simple splitter state. What you are doing in the process state can happen in this parent state.

Reply:
removed unwanted code